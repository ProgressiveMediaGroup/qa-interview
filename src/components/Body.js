// Node_Modules
import React, { Component } from 'react'
import PropTypes from 'prop-types'

class Body extends Component {
  render () {
    return (
      <main>
        <div className='strap'>
          <div className='container'>
            <h1>{this.props.heading}</h1>
            <h2>{this.props.subheading}</h2>
          </div>
        </div>
        <div className='container body-container'>
          {this.props.children}
        </div>
      </main>
    )
  }
}

Body.defaultProps = {
  heading: '',
  subheading: ''
}

Body.propTypes = {
  heading: PropTypes.string,
  subheading: PropTypes.string
}

export default Body
