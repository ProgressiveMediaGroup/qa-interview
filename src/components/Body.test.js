/* global it, describe, expect */
import React from 'react'
import { shallow, mount } from 'enzyme'
import Body from './Body'

it('renders without crashing', () => {
  shallow(<Body />)
})

it('should match snapshot', () => {
  const wrapper = shallow(<Body />)
  expect(wrapper).toMatchSnapshot()
})

describe('Testing heading', function () {
  const testHeading = 'test heading'

  // Test no defined title
  it('Test no defined heading', () => {
    const wrapper = mount(
      <Body />
    )
    expect(wrapper.prop('heading')).toBe('')
  })

  // Test title displays argument
  it('Test heading accepts argument', () => {
    const wrapper = mount(
      <Body heading={testHeading} />
    )
    expect(wrapper.prop('heading')).toBe(testHeading)
  })

  // Test title displays argument
  it('Test title displays argument', () => {
    const wrapper = mount(
      <Body heading={testHeading} />
    )
    expect(wrapper.find('h1').text()).toBe(testHeading)
  })
})

describe('Testing subheading', function () {
  const testSubHeading = 'test sub heading'

  // Test no defined title
  it('Test no defined heading', () => {
    const wrapper = mount(
      <Body />
    )
    expect(wrapper.prop('subheading')).toBe('')
  })

  // Test title displays argument
  it('Test heading accepts argument', () => {
    const wrapper = mount(
      <Body subheading={testSubHeading} />
    )
    expect(wrapper.prop('subheading')).toBe(testSubHeading)
  })

  // Test title displays argument
  it('Test title displays argument', () => {
    const wrapper = mount(
      <Body subheading={testSubHeading} />
    )
    expect(wrapper.find('h2').text()).toBe(testSubHeading)
  })
})

// Test Children are displayed (icon + text)
describe('Testing children', function () {
  it('Test rendering children correctly', () => {
    const wrapper = mount(
      <Body>Hello World</Body>
    )
    expect(wrapper.find('.body-container').text()).toBe('Hello World')
  })
})
