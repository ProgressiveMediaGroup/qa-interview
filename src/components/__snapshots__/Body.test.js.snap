// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Body
    heading=""
    subheading=""
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="strap"
>
          <div
                    className="container"
          >
                    <h1 />
                    <h2 />
          </div>
</div>,
        <div
          className="container body-container"
/>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="container"
>
            <h1 />
            <h2 />
</div>,
          "className": "strap",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h1 />,
              <h2 />,
            ],
            "className": "container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "h1",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "",
              },
              "ref": null,
              "rendered": "",
              "type": "h2",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": undefined,
          "className": "container body-container",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
    ],
    "type": "main",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="strap"
>
            <div
                        className="container"
            >
                        <h1 />
                        <h2 />
            </div>
</div>,
          <div
            className="container body-container"
/>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="container"
>
              <h1 />
              <h2 />
</div>,
            "className": "strap",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h1 />,
                <h2 />,
              ],
              "className": "container",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "",
                },
                "ref": null,
                "rendered": "",
                "type": "h1",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "",
                },
                "ref": null,
                "rendered": "",
                "type": "h2",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "container body-container",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
      ],
      "type": "main",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
